community/dbms/src/main/java/org/neo4j/commandline/dbms/MemoryRecommendationsCommand.java

```diff
@@ -0,0 +1,385 @@
+            new Bracket( 24.0, 4.5, 8 ),
+            new Bracket( 32.0, 5, 12 ),
+            new Bracket( 64.0, 6, 24 ),
+            new Bracket( 128.0, 8, 32 ),
+            new Bracket( 256.0, 12, 32 ),
+            new Bracket( 512.0, 16, 32 ),
+            new Bracket( 1024.0, 30, 32 ),
+    };
+    private static final String ARG_MEMORY = "memory";
+    private static final String ARG_PRINT_CONF = "print-conf";
+    private static final String ARG_UPDATE_CONF = "update-conf";
+    private static final String ARG_OVERWRITE = "overwrite";
+    private static final String ARG_CONF_FILE = "conf-file";
+    private final OutsideWorld outsideWorld;
+    private final Path configDir;
+
+    static long recommendOsMemory( long totalMemoryBytes )
+    {
+        Brackets brackets = findMemoryBrackets( totalMemoryBytes );
+        return brackets.recommend( Bracket::osMemory );
+    }
+
+    static long recommendHeapMemory( long totalMemoryBytes )
+    {
+        Brackets brackets = findMemoryBrackets( totalMemoryBytes );
+        return brackets.recommend( Bracket::heapMemory );
+    }
+
+    static long recommendPageCacheMemory( long totalMemoryBytes )
+    {
+        long osMemory = recommendOsMemory( totalMemoryBytes );
+        long heapMemory = recommendHeapMemory( totalMemoryBytes );
+        long recommendation = totalMemoryBytes - osMemory - heapMemory;
+        recommendation = Math.max( mebiBytes( 100 ), recommendation );
+        recommendation = Math.min( tebiBytes( 16 ), recommendation );
+        return recommendation;
+    }
+
+    private static Brackets findMemoryBrackets( long totalMemoryBytes )
+    {
+        double totalMemoryGB = ((double) totalMemoryBytes) / ((double) gibiBytes( 1 ));
+        Bracket lower = null;
+        Bracket upper = null;
+        for ( int i = 1; i < datapoints.length; i++ )
+        {
+            if ( totalMemoryGB < datapoints[i].totalMemory )
+            {
+                lower = datapoints[i - 1];
+                upper = datapoints[i];
+                break;
+            }
+        }
+        if ( lower == null )
+        {
+            lower = datapoints[datapoints.length - 1];
+            upper = datapoints[datapoints.length - 1];
+        }
+        return new Brackets( totalMemoryGB, lower, upper );
+    }
+
+    public static Arguments buildArgs()
+    {
+        return new Arguments()
+                .withArgument( new OptionalNamedArg( ARG_MEMORY, "8g", OsBeanUtil.getTotalPhysicalMemory() + "",
+                        "Recommend memory settings with respect to the given amount of memory, " +
+                        "instead of the total memory of the system running the command." ) )
+                .withArgument( new OptionalBooleanArg( ARG_PRINT_CONF, false,
+                        "Print what an updated neo4j.conf file would look like to standard out. Does not change " +
+                        "existing settings unless --overwrite is specified." ) )
```

Comment:
I wonder if you should refer the the variable name here instead of hardcoding `--overwrite`